vip-mail.php

```diff
@@ -139,6 +140,35 @@ public function filter_wp_mail_from() {
 		return 'donotreply@wpvip.com';
 	}
 
+
+	/**
+	* This function is used to handle the wp_mail failures if the sender address is rejected by the SMTP server
+	* @param WP_Error $error The WP_Error object passed by reference
+	 */
+	public function handle_wp_mail_failures( $error ) {
+		if ( defined( 'VIP_SMTP_ENABLED' ) && true === constant( 'VIP_SMTP_ENABLED' ) && isset( $error->error_data['wp_mail_failed'] ) && isset( $error->error_data['wp_mail_failed']['phpmailer_exception_code'] ) && isset( $error->errors['wp_mail_failed'] ) ) {
+			$error_data = $error->error_data['wp_mail_failed'];
+
+			// The phpmailer exception code for Sender Address rejection is 1 and we also are validating the message is matching to the one that's expected
+			if ( 1 === $error_data['phpmailer_exception_code'] && false !== strpos( $error->errors['wp_mail_failed'][0], 'Sender address rejected: not owned by user' ) ) {
+					$to          = is_array( $error_data['to'] ) ? array_map( 'esc_html', $error_data['to'] ) : esc_html( $error_data['to'] );
```

Comment:
@aagam-shah @t-wright Are you sure you want to apply `esc_html` here, and for what purpose? There are characters like `&` that are perfectly valid in email addresses that will get escaped.

https://en.wikipedia.org/wiki/Email_address#Local-part